modules/logging-loki-retention.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+:_content-type: PROCEDURE
+[id="logging-loki-retention_{context}"]
+= Enabling stream-based retention with Loki
+With Logging version 5.6 and higher, you can configure retention policies based on stream. Rules for these may be set globally, per tenant, or both. If you configure both, tenant rules apply before global rules.
+
+. To enable stream-based retention,  create or edit the `LokiStack` custom resource (CR):
+
+include::snippets/logging-create-apply-cr-snip.adoc[lines=9..12]
+
+.Example LokiStack CR
+
+[source,yaml]
+----
+  apiVersion: loki.grafana.com/v1beta1
+  kind: LokiStack
+  metadata:
+    name: lokistack-dev
+  spec:
+    size: 1x.small
+    storage:
+      secret:
+        name: test
+        type: s3
+    storageClassName: gp2
+    retention: <1>
+      deleteDelay: <2>
+    limits:
+      global: <3>
+        retentionLimits:
+          periodDays: 31
+          stream:
+            - selector:
+                name: namespace
+                type: equal
+                value: dev
+              priority: 1
+              periodDays: 1
+      tenants: <4>
+        tenanta:
+          retentionLimits:
+            periodDays: 7
+            stream:
+              - selector:
+                  name: namespace
+                  type: equal <5>
+                  value: prod
+                priority: 2
+                periodDays: 14
+              - selector:
+                  name: container
+                  type: equal
+                  value: loki
+                priority: 1
+                periodDays: 3
+        tenantb:
+          retentionLimits:
+            periodDays:
+            stream:
+              - selector:
+                  name: container
+                  type: equal
+                  value: nginx
+                priority: 1
+                periodDays: 1
```

Comment:
This is not a valid LokiStack CR, many fields are not in the lokistack CRD: https://github.com/openshift/loki/blob/release-5.6/operator/bundle/manifests/loki.grafana.com_lokistacks.yaml.

Some examples:
1. Per-Tenant stream-based retention
```
apiVersion: loki.grafana.com/v1
kind: LokiStack
metadata:
  name: lokistack-sample
  namespace: openshift-logging
spec:
  limits:
    global:
      retention:
        days: 20
    tenants:
      application:
        retention:
          days: 1
          streams:
            - days: 4
              selector: '{kubernetes_namespace_name=~"test.+"}'
      infrastructure:
        retention:
          days: 5
          streams:
            - days: 1
              selector: '{kubernetes_namespace_name=~"openshift-cluster.+"}'
  managementState: Managed
  replicationFactor: 1
  size: 1x.extra-small
  storage:
    schemas:
    - effectiveDate: "2020-10-11"
      version: v11
    secret:
      name: gcs-secret
      type: gcs
  storageClassName: standard
  tenants:
    mode: openshift-logging 
```
2. Global stream-based retention
```
 apiVersion: loki.grafana.com/v1
 kind: LokiStack
 metadata:
   name: lokistack-sample
   namespace: openshift-logging
 spec:
   limits:
     global:
       retention:
         days: 20
         streams:
         - days: 4
           priority: 1
           selector: '{kubernetes_namespace_name=~"test.+"}'
         - days: 1
           priority: 1
           selector: '{log_type="infrastructure"}'
   managementState: Managed
```